src/main/java/org/opensearch/ad/transport/ADStatsNodesTransportAction.java

```diff
@@ -11,92 +11,75 @@
 import org.opensearch.ad.stats.InternalStatNames;
 import org.opensearch.ad.task.ADTaskManager;
-import org.opensearch.cluster.service.ClusterService;
-import org.opensearch.common.inject.Inject;
 import org.opensearch.common.io.stream.StreamInput;
 import org.opensearch.monitor.jvm.JvmService;
-import org.opensearch.threadpool.ThreadPool;
-import org.opensearch.transport.TransportService;
+import org.opensearch.sdk.SDKClusterService;
+import org.opensearch.tasks.Task;
+import org.opensearch.tasks.TaskManager;
+
+import com.google.inject.Inject;
 
 /**
  *  ADStatsNodesTransportAction contains the logic to extract the stats from the nodes
  */
-public class ADStatsNodesTransportAction extends
-    TransportNodesAction<ADStatsRequest, ADStatsNodesResponse, ADStatsNodeRequest, ADStatsNodeResponse> {
-
+public class ADStatsNodesTransportAction extends TransportAction<ADStatsNodeRequest, ADStatsNodesResponse> {
     private ADStats adStats;
     private final JvmService jvmService;
     private final ADTaskManager adTaskManager;
 
+    private final SDKClusterService sdkClusterService;
+
     /**
      * Constructor
      *
-     * @param threadPool ThreadPool to use
-     * @param clusterService ClusterService
-     * @param transportService TransportService
+     * @param sdkClusterService SDK cluster Service
      * @param actionFilters Action Filters
      * @param adStats ADStats object
      * @param jvmService ES JVM Service
      * @param adTaskManager AD task manager
+     * @param taskManager Task manager
      */
     @Inject
     public ADStatsNodesTransportAction(
-        ThreadPool threadPool,
-        ClusterService clusterService,
-        TransportService transportService,
+        SDKClusterService sdkClusterService,
         ActionFilters actionFilters,
         ADStats adStats,
         JvmService jvmService,
-        ADTaskManager adTaskManager
+        ADTaskManager adTaskManager,
+        TaskManager taskManager
     ) {
-        super(
-            ADStatsNodesAction.NAME,
-            threadPool,
-            clusterService,
-            transportService,
-            actionFilters,
-            ADStatsRequest::new,
-            ADStatsNodeRequest::new,
-            ThreadPool.Names.MANAGEMENT,
-            ADStatsNodeResponse.class
-        );
+        super(ADStatsNodesAction.NAME, actionFilters, taskManager);
         this.adStats = adStats;
         this.jvmService = jvmService;
         this.adTaskManager = adTaskManager;
+        this.sdkClusterService = sdkClusterService;
     }
 
-    @Override
-    protected ADStatsNodesResponse newResponse(
-        ADStatsRequest request,
-        List<ADStatsNodeResponse> responses,
-        List<FailedNodeException> failures
-    ) {
-        return new ADStatsNodesResponse(clusterService.getClusterName(), responses, failures);
+    public ADStatsNodesResponse newResponse(List<ADStatsNodeResponse> responses, List<FailedNodeException> failures) {
```

Comment:
Let's not change the method arguments here because the tests would be affected. Old method had `ADStatsRequest` as one of the param